src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/setting/ClusterSettingsManager.java

```diff
@@ -180,6 +220,26 @@ private void callListeners(final Setting<Integer> setting, int settingValue) {
         }
     }
 
+    /**
+     * Calls all the listeners for the specified setting with the requested value.
+     *
+     * @param setting      The setting whose listeners need to be notified.
+     * @param settingValue The new value for the setting.
+     */
+    private void callStringSettingListeners(final Setting<String> setting, String settingValue) {
+        try {
+            final List<ClusterSettingListener<String>> listeners = stringSettingListenerMap.get(setting);
+            if (listeners != null) {
+                for (ClusterSettingListener<String> listener : listeners) {
+                    LOG.error("Calling string listener with value: {}", settingValue);
```

Comment:
do we want to use `DEBUG` as the log level here?

Reply:
Removed the log, I was using it only for verifying my changes.